src/app/OperationalSessionSetup.h

```diff
@@ -109,6 +109,29 @@ class OperationalDeviceProxy : public DeviceProxy
     ScopedNodeId mPeerScopedNodeId;
 };
 
+enum class CaseConnectionState : uint8_t
+{
+    Uninitialized,    // Error state: OperationalSessionSetup is useless
+    NeedsAddress,     // No address known, lookup not started yet.
+    ResolvingAddress, // Address lookup in progress.
+    HasAddress,       // Have an address, CASE handshake not started yet.
+    Connecting,       // CASE handshake in progress.
+    SecureConnected,  // CASE session established.
+    WaitingForRetry,  // No address known, but a retry is pending.  Added at
+                      // end to make logs easier to understand.
+};
+
+struct ExtendedConnectionFailureInfo
+{
+    const ScopedNodeId & peerId;
+    CHIP_ERROR error;
+    CaseConnectionState connectionState;
```

Comment:
Can we also get [CaseSession::state](https://github.com/project-chip/connectedhomeip/blob/15f54da67d08f40b18d366433c492b7c1c8dd89a/src/protocols/secure_channel/CASESession.h#L194) & an equivalent for PaseSession (seems like [mNextExpectedMessage](http://google3/third_party/connectedhomeip/next/src/protocols/secure_channel/PASESession.h), but maybe in a more readable format) in here?

Reply:
In the context of this PR, it all about establishing a CASE (Conditional Action for System Exceptions) for handling the onConnectionFailure method.  In order to manage the state for PASE (Preserve and Safeguard Exception Handling), we must approach it through a different API and within a separate PR.